include/blmc_robots/real_finger.hpp

```diff
@@ -297,77 +363,118 @@ class RealFingerDriver : public robot_interfaces::RobotDriver<
-
-        bool is_homed = home_on_index_after_negative_end_stop(TORQUE_RATIO,
-                                                              home_offset_rad);
+        bool is_homed = home_on_index_after_negative_end_stop(
+            params.torque_ratio, home_offset_rad_);
 
         if (is_homed)
         {
-            bool reached_goal = move_to_position(
-                starting_position_rad, CONTROL_GAIN_KP, CONTROL_GAIN_KD,
-                POSITION_TOLERANCE_RAD, MOVE_TIMEOUT);
+            bool reached_goal = move_to_position(starting_position_rad_,
+                                                 params.control_gain_kp,
+                                                 params.control_gain_kd,
+                                                 params.position_tolerance_rad,
+                                                 params.move_timeout);
             if (!reached_goal)
             {
                 rt_printf("Failed to reach goal, timeout exceeded.\n");
             }
         }
     }
 
-private:
+protected:
+    /**
+     * \brief Offset between home position and zero.
+     *
+     * Defined such that the zero position is at the negative end stop (for
+     * compatibility with old homing method).
+     */
+    Vector home_offset_rad_ = Vector::Zero();
+
+    //! \brief Start position to which the robot moves after homing.
+    Vector starting_position_rad_ = Vector::Zero();
+
+    //! \brief D-gain to dampen velocity.  Set to zero to disable damping.
+    Vector safety_kd_ = Vector::Zero();
+
     /// todo: this should probably go away
-    double max_torque_;
+    double max_torque_Nm_;
 
-    BlmcJointModules<3> joint_modules_;
+    BlmcJointModules<N_JOINTS> joint_modules_;
     MotorBoards motor_boards_;
 
-    /// todo: this should probably go away
+public:
+    Vector get_max_torques() const { return max_torque_Nm_ * Vector::Ones(); }
+};
 
+class RealFingerDriver : public NJointBlmcRobotDriver<3, 2>
+{
 public:
-    Vector get_max_torques() const { return max_torque_ * Vector::Ones(); }
+    RealFingerDriver(const std::string &can_0, const std::string &can_1)
+        : NJointBlmcRobotDriver<3, 2>({can_0, can_1},
+                                  {.max_current_A = 2.0,
+                                   .torque_constant_NmpA = 0.02,
+                                   .gear_ratio = 9.0},
+                                  0.003,
+                                  0.005)
+    {
+        initialize();
+    }
+
+    // TODO do we need this constructor?
+    // RealFingerDriver(const MotorBoards &motor_boards)
+    //    : NJointBlmcRobotDriver<3, 2>(motor_boards, 0.003, 0.005)
+    //{
+    //    initialize();
+    //}
+
+protected:
+    // this is needed to be able to overload the method above.
+    using NJointBlmcRobotDriver<3, 2>::calibrate;
+
+    void initialize()
+    {
+        safety_kd_ << 0.08, 0.08, 0.04;
+        home_offset_rad_ << -0.54, -0.17, 0.0;
+        starting_position_rad_ << 1.5, 1.5, 3.0;
```

Comment:
i am not super comfortable with setting these parameters in this way, we might easily forget setting them when creating some other class. (which would be particularly dangerous for the safety_kd_). therefore, i think i would require them as arguments in the constructor of NJointBlmcRobotDriver.

Reply:
The advantage of this approach is that all values are explicitly names, compared to a list of anonymous values when calling the constructor:
```
NJointBlmcRobotDriver(...
                      Vector(0.08, 0.08, 0.04),
                      Vector(-0.54, -0.17, 0.0),
                      Vector(1.5, 1.5, 3.0))
```
But I understand your concern. I'll try to find a way to pass it to the c'tor while still having names related to the values.

Reply:
I only added safety_kd to the constructor now to not overload it too much.  Also home offset and initial position are not so critical and zero is a reasonable default there.

Reply:
makes sense!